docs/website/documentation/faq.md

```diff
@@ -20,3 +20,29 @@ Supplying the answers are a TODO.
 ## How do I supply answers?
 
 Click the "Edit this Page" link in the bottom right.
+
+
+# Common Problems:
+
+## java.lang.OutOfMemoryError: unable to create new native thread
+
+You could encounter this error when running with many entities.
+
+Please **increase the ulimit** if you see such error:
+
+On the VM running Apache Brooklyn, we recommend ensuring nproc and nofile are reasonably high (e.g. higher than 1024, which is often the default).
+
+Run `ulimit -a` to see the current limits.
+
+To increase the limits, run `sudo vi /etc/security/limits.conf` and add (if it is "brooklyn" user running Apache Brooklyn):
+
+    brooklyn           soft    nproc           16384
+    brooklyn           hard    nproc           16384
+    brooklyn           soft    nofile          16384
+    brooklyn           hard    nofile          16384
+
+Then apply these changes by running `sudo sysctl -p`.
```

Comment:
limits.conf is only read during login shells.

`sysctl -p` only enforces re-read of /etc/sysctl.conf